### PR TITLE
feat: add vrops integration in vra

### LINKED ISSUE #44: Add support for adding the vRealize Operations integration in vRealize Automation
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Develop PowerShell cmdlet(s) to enable the addition of a vRealize Operations integration in vRealize Automation.

### Use Case(s)

Provides an automation path for the procedure.

### Potential Configuration

To be determined during implementation.

### References

Guidance: [Integrate vRealize Automation with vRealize Operations Manager](https://docs.vmware.com/en/VMware-Cloud-Foundation/services/vcf-private-cloud-automation-v1/GUID-E2D50808-3946-4B78-8810-04A8B1F98586.html)

### PR DESCRIPTION
**Pull request by slakshmanan2706**
**Summary of Pull Request**

Adds support for integrating vRealize Automation with vRealize Operations Manager

Adds the following cmdlets to support the same.

- `New-vRAvROPSIntegrationItem`
- `Add-vRAIntegrationItem`
- `Test-vRAIntegrationItem`
- `Remove-vRAIntegrationItem`
- `Undo-vRAvROPsIntegrationItem`

Signed-off-by: Lakshmanan Shanmugam <lshanmugam@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes #44  

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -26342,6 +26342,279 @@ Function Remove-vRAUserServiceRole {
...
+        if (Test-VCFConnection -server $server) {
+            if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+                if (($vcfVraDetails = Get-vRAServerDetail -fqdn $server -username $user -password $pass)) {
+                    if (Test-vRAConnection -server $vcfVraDetails.loadBalancerFqdn) {
+                        if (Test-vRAAuthentication -server $vcfVraDetails.loadBalancerFqdn -user $vraUser -pass $vraPass) {
+                            if (($vcfVropsDetails = Get-vROPsServerDetail -fqdn $server -username $user -password $pass)) {
+                                if (Test-vROPSConnection -server $vcfVropsDetails.loadBalancerFqdn) {
+                                    if (Test-vROPSAuthentication -server $vcfVropsDetails.loadBalancerFqdn -user $vcfVropsDetails.adminUser -pass $vcfVropsDetails.adminPass) {
+                                        if ($null -eq (Get-vRAIntegrationDetail -integrationType "vrops"  -integrationName $vropsIntegrationName -getIntegrationID ) ) {
+                                            Write-Output "vRealize Operations Manager Integration with name '$vropsIntegrationName' not found..."
```
[the remark this answers is not in the document]
Comment (slakshmanan2706) [reply PowerValidatedSolutions.psm1:26596]: Done

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -26342,6 +26342,279 @@ Function Remove-vRAUserServiceRole {
...
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    Catch {
```
[the remark this answers is not in the document]
Comment (slakshmanan2706) [reply PowerValidatedSolutions.psm1:26612]: Done

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -26342,6 +26342,279 @@ Function Remove-vRAUserServiceRole {
...
+    Try {
+        $vraapiVersion = "apiVersion=" + (Get-vRAAPIVersion)
+        $uri = "https://$vraAppliance/iaas/api/integrations/$integrationId" + "?$vraapiVersion"     
+        $response = Invoke-RestMethod -Method Delete -Uri $uri -Headers $vraHeaders
+        Start-Sleep 5
+        $id = $response.selfLink
+        $uri = "https://$vraAppliance$id"
+        $response = Invoke-RestMethod -Method Get -Uri $uri -Headers $vraHeaders
+        if ( $response.status -eq "FAILED") {
+            Write-Host "Error "$response.message
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:26545]: Chang Write-Host to Write-Error

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Some minor changes requested. 😄

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
@slakshmanan2706 - the changes look **great**, thank you!

One last minor item I noticed in this review - could you update the example passwords to conform to the other examples?

- from "VMw@re123!" to "VMw@re1!"
- from "VMware123!" to "VMw@re1!"

Thanks!
Ryan

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

**Pull request merged**